libixx/src/test/mod.rs

```diff
@@ -14,55 +14,70 @@ fn test() {
   index.push(0, "programs.home-manager.enable");
   index.push(0, "services.home-manager.autoUpgrade.enable");
   index.push(0, "services.home-manager.autoUpgrade.frequency");
+  index.push(1, "home.enableDebugInfo");
 
   assert_eq!(
     index.search(None, "ho*auto", 10).unwrap(),
     vec![
+      (7, 0, "services.home-manager.autoUpgrade.enable".to_string()),
       (
-        7usize,
-        "services.home-manager.autoUpgrade.enable".to_string()
-      ),
-      (
-        8usize,
+        8,
+        0,
         "services.home-manager.autoUpgrade.frequency".to_string()
       )
     ]
   );
 
   assert_eq!(
     index.search(None, "ho*auto*ena", 10).unwrap(),
-    vec![(
-      7usize,
-      "services.home-manager.autoUpgrade.enable".to_string()
-    )]
+    vec![(7, 0, "services.home-manager.autoUpgrade.enable".to_string())]
   );
 
   assert_eq!(
     index.search(None, "ho*en*Nix", 10).unwrap(),
-    vec![(1usize, "home.enableNixpkgsReleaseCheck".to_string())]
+    vec![(1, 0, "home.enableNixpkgsReleaseCheck".to_string())]
   );
 
   assert_eq!(
     index.search(None, "ho*en*Nix*Rel*Che", 10).unwrap(),
-    vec![(1usize, "home.enableNixpkgsReleaseCheck".to_string())]
+    vec![(1, 0, "home.enableNixpkgsReleaseCheck".to_string())]
   );
 
   assert_eq!(
     index.search(None, "enablenixpkgsreleasecheck", 10).unwrap(),
-    vec![(1usize, "home.enableNixpkgsReleaseCheck".to_string())]
+    vec![(1, 0, "home.enableNixpkgsReleaseCheck".to_string())]
   );
 
   // TEST scopes
   assert_eq!(
     index
       .search(Some(0), "enablenixpkgsreleasecheck", 10)
       .unwrap(),
-    vec![(1usize, "home.enableNixpkgsReleaseCheck".to_string())]
+    vec![(1, 0, "home.enableNixpkgsReleaseCheck".to_string())]
   );
   assert_eq!(
     index
       .search(Some(1), "enablenixpkgsreleasecheck", 10)
       .unwrap(),
     vec![]
   );
+
+  // TEST options with same name in different scopes
+  assert_eq!(
+    index.search(None, "ho*debug", 10).unwrap(),
+    vec![
+      (0, 0, "home.enableDebugInfo".to_string()),
+      (9, 1, "home.enableDebugInfo".to_string())
+    ]
+  );
+
+  assert_eq!(
```

Comment:
Maybe we should also test a none match?

Reply:
What do you mean "non match"? We compare the vector. That means that additional elements would also fail the assertion.